src/server/detail/snapshot_storage.h

```diff
@@ -115,7 +122,8 @@ class LinuxWriteWrapper : public io::Sink {
   off_t offset_ = 0;
 };
 
-void SubstituteFilenameTsPlaceholder(fs::path* filename, std::string_view replacement);
+void SubstituteFilenamePlaceholders(fs::path* filename, std::string_view ts, std::string_view year,
+                                    std::string_view month, std::string_view day);
```

Comment:
This has become a pretty long list of variables of the same type, and order mistakes could happen.

Consider creating a struct along the lines of
```cpp
struct FilenameSubstitutions {
  string_view ts;
  string_view year;
  string_view month;
  string_view day;
};
```
And then calling this function could be more explicit:
`SubstituteFilenamePlaceholders(file, {.ts = ..., .year = ..., .month = ..., .day = ...});`
It'd be easier to spot mistakes like that imo